syntax = "proto3";


option go_package = "pb/game";

import "proto/common/db_base.proto";

message redis_node_list_db
{
  repeated redis_node_db redis_info = 1; 
}

message redis_node_db
{
  option(OptionTableName)  = "redis_node_db";
  option(OptionPrimaryKey) = "id";
  option(OptionAutoIncrementKey) = "id";
  uint32 id = 1;
  string ip = 2;
  uint32 port = 3;
  uint32 zone_id = 4;
}

message NodeInfoRequest
{
  uint32 zone_id = 1;
  uint32 node_type = 2;
}

message NodeInfoResponse
{
  redis_node_list_db redis_info = 1;
  uint32 node_id = 2;
  uint64 lease_id = 3; //租约
}


// 请求分配 ID
message GetIDRequest 
{
  string client_name = 1;
  uint32 node_type = 2; // 服务器类型，例如 "gate" 或 "scene"
}

// 返回的 ID
message GetIDResponse 
{
  uint64 id = 1;  // 分配的唯一 ID
  uint64 lease_id = 2; //租约
}

// 请求释放 ID
message ReleaseIDRequest 
{
  uint64 id = 1;  // 要释放的 ID
  uint32 node_type = 2;  // 服务器类型
  uint64 lease_id = 3; //租约
}

// 返回的响应
message ReleaseIDResponse 
{
  bool success = 1;  // 是否成功回收 ID
}

// 续租 ID
message RenewLeaseIDRequest 
{
  uint64 lease_id = 1; //租约
}

// 
message RenewLeaseIDResponse 
{
}

service DeployService 
{
  rpc GetNodeInfo (NodeInfoRequest) returns (NodeInfoResponse);
  rpc GetID (GetIDRequest) returns (GetIDResponse);
  rpc ReleaseID (ReleaseIDRequest) returns (ReleaseIDResponse);
  rpc RenewLease (RenewLeaseIDRequest) returns (RenewLeaseIDResponse);
}
